util/wait_group_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitGroupTimeout(t *testing.T) {
+	t.Parallel()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		wg.Done()
+	}()
+	timeout := time.Millisecond * 10
+	rs := WaitTimeout(wg, timeout)
+	assert.Equal(t, false, rs)
```

Comment:
If the goroutine takes more than 10ms, this result become undecidable

Reply:
Any suggestion to test this kind of timeout func?

Reply:
refactored and removed.